services/ppac/src/main/java/app/coronawarn/datadonation/services/ppac/commons/PpaDataRequestConverter.java

```diff
@@ -66,10 +67,13 @@ protected ExposureWindowsAtTestRegistration convertToExposureWindowAtTestRegistr
     PPAExposureWindow exposureWindow = newExposureWindow.getExposureWindow();
     Set<ScanInstancesAtTestRegistration> scanInstancesAtTestRegistration =
         convertToScanInstancesAtTestRegistrationEntities(newExposureWindow);
-    return new ExposureWindowsAtTestRegistration(null, null, getLocalDateFor(exposureWindow.getDate()),
-        exposureWindow.getReportTypeValue(), exposureWindow.getInfectiousnessValue(),
-        exposureWindow.getCalibrationConfidence(), newExposureWindow.getTransmissionRiskLevel(),
-        newExposureWindow.getNormalizedTime(), scanInstancesAtTestRegistration, afterTestRegistration);
+    if (exposureWindow != null) {
+      return new ExposureWindowsAtTestRegistration(null, null, getLocalDateFor(exposureWindow.getDate()),
+          exposureWindow.getReportTypeValue(), exposureWindow.getInfectiousnessValue(),
+          exposureWindow.getCalibrationConfidence(), newExposureWindow.getTransmissionRiskLevel(),
+          newExposureWindow.getNormalizedTime(), scanInstancesAtTestRegistration, afterTestRegistration);
+    }
+    return null;
```

Comment:
if you return `null` here, you will also have `null` in the set, right? But it shouldn't be there, right?
```java
return exposureWindows.stream()
          .map(newWindow -> convertToExposureWindowAtTestRegistration(newWindow, afterTestRegistration))
          .collect(Collectors.toSet());
```

Reply:
Yes, you are right. I will filter out the nulls from the sets, which would mean that the exposureWindow was null in the payload and we have nothing to store in DB.